### DISCUSSION TITLE
Appropriate additions for the monorepo preset?

### DISCUSSION THREAD

**Discussion by TobiX** [renovatebot/renovate#14297] [Request Help] [answered]
While drafting https://github.com/renovatebot/renovate/pull/14296 I was asking myself what was appropriate for the monorepo preset:

- Technically, no presets for Maven dependencies are needed, since both Gradle & Maven provide mechanisms to move common version numbers into a variable and only update that.
- On the other hand, there are many "non-ideally" configured projects out there, which duplicate version numbers for related dependencies in their `pom.xml` or `build.gradle` - Those project get a benefit out of the monorepo definitions, since updates of related dependencies are now grouped into one PR

So my question: Should I continue sending PRs for the monorepo preset for bigger Java projects, which only benefits projects which aren't using best practices? 😉

**Comment by rarkins** [comment, marked as answer]
We have a brief guide here: https://github.com/renovatebot/renovate/blob/main/docs/development/creating-editing-renovate-presets.md

You can pay particular attention to the difference between group:monorepo and group:recommended.

In future, we hope to default to grouping dependencies by source repo instead of dependency name by default, meaning "automatic" monorepo grouping by default without needing so many manual presets in this repo. Until then, PRs are welcome to add more, if they benefit the wider community.

**Reply by TobiX** [reply]
Thanks for the clarification!
